storm-client/src/jvm/org/apache/storm/daemon/GrouperFactory.java

```diff
@@ -137,7 +137,7 @@ public void prepare(WorkerTopologyContext context, GlobalStreamId stream, List<I
     public static class FieldsGrouper implements CustomStreamGrouping {
 
         private Fields outFields;
-        private List<Integer> targetTasks;
+        private ArrayList<List<Integer> > targetTasks;
```

Comment:
There is a minor diff.. In particular here I am critically counting on the stronger guarantees of const indexing time that is not provided by List<>.